roles/openshift_master/tasks/system_container.yml

```diff
@@ -17,20 +17,62 @@
   changed_when: "'Extracting' in update_result.stdout"
   when:
     - ("master" in result.stdout)
-    - (openshift.common.version is defined) and (openshift.common.version == openshift_version) | bool
+    - l_is_same_version
+    - not l_is_ha
 
 - name: Uninstall Master system container package
   command: >
     atomic uninstall {{ openshift.common.service_type }}-master
   failed_when: False
   when:
     - ("master" in result.stdout)
-    - (openshift.common.version is not defined) or (openshift.common.version != openshift_version) | bool
+    - l_is_same_version
+    - not l_is_same_version
+    - not l_is_ha
 
 - name: Install Master system container package
   command: >
     atomic install --system --name={{ openshift.common.service_type }}-master {{ openshift.common.system_images_registry }}/{{ openshift.master.master_system_image }}:{{ openshift_image_tag }}
   when:
-    - (openshift.common.version is not defined) or (openshift.common.version != openshift_version) or ("master" not in result.stdout) | bool
+    - not l_is_same_version or ("master" not in result.stdout) | bool
+    - not l_is_ha
```

Comment:
@giuseppe that is why I switched up the logic, `and 'master' in result.stdout` should be the same as `or 'master' not in result.stdout`